### PR TITLE
Stress histogram refactor to use Google Test

### PR DESCRIPTION
**Pull request by martinmine**
This splits up the file test/stressHistogram.cxx into smaller, more independent files using Google Test that is more maintainable than the existing code.

### CODE REVIEW

**Comment by martinmine** [comment]
Seems that something went wrong during the rebase. Will have a look tomorrow.

File: hist/hist/test/stress/StressHistogramGlobal.cxx [hunk 1]
```diff
@@ -0,0 +1,529 @@
+// @(#)root/test:$Id$
+// Authors: David Gonzalez Maline November 2008
+//          Martin Storø Nyfløtt  June 2017
+
+#include <sstream>
```
Comment (vgvassilev) [inline hist/hist/test/stress/StressHistogramGlobal.cxx]: Includes should be ordered by generality. libc, libc++ includes go last.

File: hist/hist/test/stress/StressHistogramGlobal.cxx [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+#include <cmath>
+
+#include "TH2.h"
+#include "TH3.h"
+#include "THn.h"
+#include "THnSparse.h"
+
+#include "TProfile.h"
+#include "TF1.h"
+#include "HFitInterface.h"
```
Comment (vgvassilev) [inline hist/hist/test/stress/StressHistogramGlobal.cxx]: We usually order by name.

File: hist/hist/test/stress/add/testAdd.cxx [hunk 3]
```diff
@@ -0,0 +1,94 @@
+// @(#)root/test:$Id$
+// Authors: David Gonzalez Maline November 2008
+//          Martin Storø Nyfløtt  June 2017
+
+#include <sstream>
```
Comment (vgvassilev) [inline hist/hist/test/stress/add/testAdd.cxx]: Likewise.

**Comment by martinmine** [comment]
Thanks for your comments @vgvassilev, I'll address the header order shortly. As a side-note, could this be something we could make clang-format or one of those tools check for?

**Comment by vgvassilev** [comment]
@martinmine, sure IIRC that's a clang-tidy option that we could enable. Could you investigate?

**Comment by martinmine** [comment]
@vgvassilev [Yep, there is.](https://clang.llvm.org/extra/clang-tidy/checks/llvm-include-order.html) I'll submit another PR for this.

**Comment by martinmine** [comment]
@vgvassilev I've fixed your comment regarding the header ordering in 30ece34 and 3428e14

**Comment by martinmine** [comment]
Rebased after a merge conflict

**Comment by lmoneta** [comment]
This is a very nice contribution, but it brings a very small advantage to a test that is already very modular, because composed by very small functions, which can be run all independently and from the ROOT prompt. 
I am sorry but I will have to close this PR. 
 Lorenzo

**Pull request closed without merging**